src/server/game/Loot/LootMgr.cpp

```diff
@@ -412,88 +412,54 @@ LootItem::LootItem(LootStoreItem const& li)
     groupid = li.groupid;
 }
 
-// Basic checks for player/item compatibility - if false no chance to see the item in the loot
 bool LootItem::AllowedForPlayer(Player const* player, ObjectGuid source) const
 {
-    ItemTemplate const* pProto = sObjectMgr->GetItemTemplate(itemid);
-    if (!pProto)
-    {
-        return false;
-    }
-
-    if (DisableMgr::IsDisabledFor(DISABLE_TYPE_LOOT, itemid, nullptr))
-    {
-        return false;
-    }
+    return AllowedForPlayer(player, false, source);
+}
 
-    bool isMasterLooter = player->GetGroup() && player->GetGroup()->GetMasterLooterGuid() == player->GetGUID();
-    bool itemVisibleForMasterLooter = !needs_quest && (!follow_loot_rules || !is_underthreshold);
+bool LootItem::AllowedForPlayer(Player const* player, bool isGivenByMasterLooter) const
```

Comment:
This function is not called to decide for Shadowshard fragment (quest item) when deciding if loot should be visible to a player

Reply:
When debugging I found this returns no items if the loot master if not eligible. If loot master there should be items. There's a special case for other items above where non-items are added for loot master, but this isn't done here. Maybe the fix can be done here

- https://github.com/EricksOliveira/azerothcore-wotlk/blob/22490b40bc4ad61fcab54acab54e90de46f1be47/src/server/game/Loot/LootMgr.cpp#L1128